app_dart/test/request_handlers/refresh_chromebot_status_test.dart

```diff
@@ -63,12 +104,65 @@ void main() {
         return Future<Map<LuciBuilder, List<LuciTask>>>.value(luciTasks);
       });
 
-      expect(task.status, isNot(Task.statusNew));
-      await tester.get(handler);
       expect(task.status, Task.statusNew);
+      branchHttpClient.request.response.body = branchRegExp;
+      await tester.get(handler);
+      expect(task.status, Task.statusSucceeded);
+    });
+
+    test('updates datastore entry for task - multiple branches', () async {
+      final Commit commitMaster = Commit(
+          key: config.db.emptyKey.append(Commit, id: 'abc'),
+          sha: 'abc',
+          branch: 'master');
+      final Commit commitOther = Commit(
+          key: config.db.emptyKey.append(Commit, id: 'def'),
+          sha: 'def',
+          branch: 'flutter-0.0-candidate.0');
+      final Task taskMaster = Task(
+          key: commitMaster.key.append(Task, id: 123), status: Task.statusNew);
+      final Task taskOther = Task(
+          key: commitOther.key.append(Task, id: 456), status: Task.statusNew);
+      config.db.values[commitMaster.key] = commitMaster;
+      config.db.values[commitOther.key] = commitOther;
+      config.db.values[taskMaster.key] = taskMaster;
+      config.db.values[taskOther.key] = taskOther;
+      githubBranches = <String>['master', 'flutter-0.0-candidate.0'];
+
+      final Map<LuciBuilder, List<LuciTask>> luciTasks =
+          Map<LuciBuilder, List<LuciTask>>.fromIterable(
+        await LuciBuilder.getBuilders(config),
+        key: (dynamic builder) => builder as LuciBuilder,
+        value: (dynamic builder) => <LuciTask>[
+          const LuciTask(
+              commitSha: 'abc',
+              ref: 'refs/heads/master',
+              status: Task.statusSucceeded),
+          const LuciTask(
+              commitSha: 'def',
+              ref: 'refs/heads/flutter-0.0-candiate.0',
+              status: Task.statusSucceeded)
+        ],
+      );
+      when(mockLuciService.getRecentTasks(
+              repo: 'flutter', requireTaskName: true))
+          .thenAnswer((Invocation invocation) {
+        return Future<Map<LuciBuilder, List<LuciTask>>>.value(luciTasks);
+      });
+
+      expect(taskMaster.status, Task.statusNew);
+      expect(taskOther.status, Task.statusNew);
+      branchHttpClient.request.response.body = branchRegExp;
+      await tester.get(handler);
+      expect(taskMaster.status, Task.statusSucceeded);
+      expect(taskOther.status, Task.statusSucceeded);
     });
   });
 }
 
 // ignore: must_be_immutable
 class MockLuciService extends Mock implements LuciService {}
+
+class MockGitHub extends Mock implements GitHub {}
```

Comment:
These mocks could also be moved to the test utils file to remove the need for creating them in each file

Reply:
Ditto.